web/modules/custom/bnf/bnf_client/bnf_client.module

```diff
@@ -48,31 +49,66 @@ function bnf_client_form_alter(array &$form, FormStateInterface $form_state, str
   // Let's hide the publishing button when BNF is checked, as the value will
   // be ignored and the node will be published regardless.
   $form['status']['#states']['invisible'] =
-    [':input[name="bnf"]' => ['checked' => TRUE]];
+    [':input[name="bnf_export"]' => ['checked' => TRUE]];
 
-  $form['bnf'] = [
+  // Getting the BNF state, if it exists.
+  $state = BnfStateEnum::None;
+
+  if ($node->hasField(BnfStateEnum::FIELD_NAME) && !$node->get(BnfStateEnum::FIELD_NAME)->isEmpty()) {
+    /** @var \Drupal\enum_field\Plugin\Field\FieldType\EnumItemList $state_field */
+    $state_field = $node->get(BnfStateEnum::FIELD_NAME);
+    $states = $state_field->enums();
```

Comment:
Doesn't `$state_field->enum` work? `FieldItemList` has a magic getter that'll get the named property from the first item.

I would think that `$state = node->get(BnfStateEnum::FIELD_NAME)->enum ?? BnfStateEnum::None;` ought to work.

Reply:
Nope, neither `->enum` or `->enum()` works. Only `->enums()`

Reply:
Weird, it should work, according to the module page:

https://www.drupal.org/project/enum_field